decred.go

```diff
@@ -19,110 +21,108 @@ import (
 )
 
 const (
-	// defaultAddressTimeout defines the duration to wait
-	// for new addresses.
+	// defaultAddressTimeout defines the duration to wait for new addresses.
 	defaultAddressTimeout = time.Minute * 10
 
-	// defaultNodeTimeout defines the timeout time waiting for
-	// a response from a node.
+	// defaultNodeTimeout defines the timeout on responses from a node.
 	defaultNodeTimeout = time.Second * 3
 )
 
-var (
-	amgr *Manager
-	wg   sync.WaitGroup
-)
+var amgr *Manager
+
+func testPeer(ctx context.Context, ip net.IP, netParams *chaincfg.Params) {
+	onaddr := make(chan struct{}, 1)
+	verack := make(chan struct{}, 1)
```

Comment:
@jholdstock made a good find with these channels being used improperly, so I took one extra step and eliminated the closure.  The `creep` function is really concise now.